pubsub/client.go

```diff
@@ -85,37 +75,27 @@ type Client struct {
 	// The current state that the client is in.
 	state     *fsm.Machine
 	stateLock *sync.Mutex
-	// Connection params we're subbed to.
-	conn ConnectionParam
+	// Connection we're subbed to.
+	conn *conn.Connector
```

Comment:
I'd rather code to an interface here so that if people want to provide their own connectors, they can. Maybe code to the Redigo Pool; that's a popular package, and connection pooling support would be good.